### ISSUE TITLE
数据加载阶段程序挂了

### ISSUE THREAD

**Issue by jiezhangGt** [LianjiaTech/BELLE#313]
你好，我在45G的8卡上训练bloom-7b, 数据上加了一些中英文单语进去，共计6000w左右；然后数据在加载到30多万的时候就崩了。。。请问这个一般是什么原因导致的啊？

`length of train_dataset(after get_train_data):  59719579
100%|██████████| 1/1 [00:00<00:00, 610.35it/s]
Found cached dataset json (/workspace/BELLE/train/output/data_files/json/default-0dad753e06ba0704/0.0.0/fe5dd6ea2639a6df622901539cb550cf8797e5a6b2dd7af1cf934bed8e233e6e)
Dataset({
    features: ['instruction', 'input', 'output'],
    num_rows: 59719579
})
length of train_dataset(after get_train_data):  59719579
100%|██████████| 1/1 [00:00<00:00, 538.49it/s]
Dataset({
    features: ['instruction', 'input', 'output'],
    num_rows: 59719579
})
length of train_dataset(after get_train_data):  59719579
Found cached dataset json (/workspace/BELLE/train/output/data_files/json/default-0dad753e06ba0704/0.0.0/fe5dd6ea2639a6df622901539cb550cf8797e5a6b2dd7af1cf934bed8e233e6e)
100%|██████████| 1/1 [00:00<00:00, 568.26it/s]
Dataset({
    features: ['instruction', 'input', 'output'],
    num_rows: 59719579
})
length of train_dataset(after get_train_data):  59719579
  1%|          | 329183/59719579 [12:45<58:16:54, 283.06example/s][2023-04-26 01:59:43,468] [INFO] [launch.py:428:sigkill_handler] Killing subprocess 629
[2023-04-26 01:59:43,470] [INFO] [launch.py:428:sigkill_handler] Killing subprocess 630
  1%|          | 330273/59719579 [12:49<65:53:11, 250.39example/s][2023-04-26 01:59:47,484] [INFO] [launch.py:428:sigkill_handler] Killing subprocess 631
  1%|          | 331325/59719579 [12:53<60:07:50, 274.35example/s][2023-04-26 01:59:51,378] [INFO] [launch.py:428:sigkill_handler] Killing subprocess 632
[2023-04-26 01:59:51,379] [INFO] [launch.py:428:sigkill_handler] Killing subprocess 633
  1%|          | 332434/59719579 [12:57<56:55:09, 289.82example/s][2023-04-26 01:59:55,313] [INFO] [launch.py:428:sigkill_handler] Killing subprocess 634
  1%|          | 333582/59719579 [13:01<57:22:32, 287.51example/s][2023-04-26 01:59:59,167] [INFO] [launch.py:428:sigkill_handler] Killing subprocess 637
  1%|          | 334685/59719579 [13:05<62:28:33, 264.03example/s][2023-04-26 02:00:03,101] [INFO] [launch.py:428:sigkill_handler] Killing subprocess 778
[2023-04-26 02:00:07,035] [ERROR] [launch.py:434:sigkill_handler] ['/opt/conda/bin/python3', '-u', 'main.py', '--local_rank=7', '--sft_only_data_path', '/workspace/BELLE/train/1_task_nmt/data/train.json', '--model_name_or_path', '/workspace/BELLE/train/data/model_name_or_path/models--bigscience--bloom-7b1/snapshots/e83e90ba86f87f74aa2731cdab25ccf33976bd66', '--per_device_train_batch_size', '1', '--per_device_eval_batch_size', '1', '--max_seq_len', '384', '--learning_rate', '5e-6', '--weight_decay', '0.0001', '--num_train_epochs', '5', '--gradient_accumulation_steps', '8', '--lr_scheduler_type', 'cosine', '--num_warmup_steps', '100', '--seed', '1234', '--gradient_checkpointing', '--zero_stage', '2', '--deepspeed', '--output_dir', 'output', '--data_output_path', 'output/data_files'] exits with return code = -9`

**Comment by ljg-lixufeng** [comment]
我也是，报错信息都没有

**Comment by xianghuisun** [comment]
> 你好，我在45G的8卡上训练bloom-7b, 数据上加了一些中英文单语进去，共计6000w左右；然后数据在加载到30多万的时候就崩了。。。请问这个一般是什么原因导致的啊？
> 
> `length of train_dataset(after get_train_data): 59719579 100%|██████████| 1/1 [00:00<00:00, 610.35it/s] Found cached dataset json (/workspace/BELLE/train/output/data_files/json/default-0dad753e06ba0704/0.0.0/fe5dd6ea2639a6df622901539cb550cf8797e5a6b2dd7af1cf934bed8e233e6e) Dataset({ features: ['instruction', 'input', 'output'], num_rows: 59719579 }) length of train_dataset(after get_train_data): 59719579 100%|██████████| 1/1 [00:00<00:00, 538.49it/s] Dataset({ features: ['instruction', 'input', 'output'], num_rows: 59719579 }) length of train_dataset(after get_train_data): 59719579 Found cached dataset json (/workspace/BELLE/train/output/data_files/json/default-0dad753e06ba0704/0.0.0/fe5dd6ea2639a6df622901539cb550cf8797e5a6b2dd7af1cf934bed8e233e6e) 100%|██████████| 1/1 [00:00<00:00, 568.26it/s] Dataset({ features: ['instruction', 'input', 'output'], num_rows: 59719579 }) length of train_dataset(after get_train_data): 59719579 1%| | 329183/59719579 [12:45<58:16:54, 283.06example/s][2023-04-26 01:59:43,468] [INFO] [launch.py:428:sigkill_handler] Killing subprocess 629 [2023-04-26 01:59:43,470] [INFO] [launch.py:428:sigkill_handler] Killing subprocess 630 1%| | 330273/59719579 [12:49<65:53:11, 250.39example/s][2023-04-26 01:59:47,484] [INFO] [launch.py:428:sigkill_handler] Killing subprocess 631 1%| | 331325/59719579 [12:53<60:07:50, 274.35example/s][2023-04-26 01:59:51,378] [INFO] [launch.py:428:sigkill_handler] Killing subprocess 632 [2023-04-26 01:59:51,379] [INFO] [launch.py:428:sigkill_handler] Killing subprocess 633 1%| | 332434/59719579 [12:57<56:55:09, 289.82example/s][2023-04-26 01:59:55,313] [INFO] [launch.py:428:sigkill_handler] Killing subprocess 634 1%| | 333582/59719579 [13:01<57:22:32, 287.51example/s][2023-04-26 01:59:59,167] [INFO] [launch.py:428:sigkill_handler] Killing subprocess 637 1%| | 334685/59719579 [13:05<62:28:33, 264.03example/s][2023-04-26 02:00:03,101] [INFO] [launch.py:428:sigkill_handler] Killing subprocess 778 [2023-04-26 02:00:07,035] [ERROR] [launch.py:434:sigkill_handler] ['/opt/conda/bin/python3', '-u', 'main.py', '--local_rank=7', '--sft_only_data_path', '/workspace/BELLE/train/1_task_nmt/data/train.json', '--model_name_or_path', '/workspace/BELLE/train/data/model_name_or_path/models--bigscience--bloom-7b1/snapshots/e83e90ba86f87f74aa2731cdab25ccf33976bd66', '--per_device_train_batch_size', '1', '--per_device_eval_batch_size', '1', '--max_seq_len', '384', '--learning_rate', '5e-6', '--weight_decay', '0.0001', '--num_train_epochs', '5', '--gradient_accumulation_steps', '8', '--lr_scheduler_type', 'cosine', '--num_warmup_steps', '100', '--seed', '1234', '--gradient_checkpointing', '--zero_stage', '2', '--deepspeed', '--output_dir', 'output', '--data_output_path', 'output/data_files'] exits with return code = -9`

内存不足

**Comment by jiezhangGt** [comment]
> 

我的机器是单机8卡，45G显存，一般来说不会内存不够？

**Comment by jiezhangGt** [comment]
![Uploading image.png…]()
内存是不是太小了？377G

**Comment by LJL36** [comment]
> > 
> 
> 我的机器是单机8卡，45G显存，一般来说不会内存不够？会不会是docker没给全部的机器的内存？

您好，我这边也是375G内存，加载数据跟你报了一样的错误，我用的还是0.5M的数据，您这边解决了吗？
